azuredevops/internal/service/wiki/resource_wiki.go

```diff
@@ -0,0 +1,199 @@
+	uuidProject, err := uuid.Parse(d.Get("project_id").(string))
+	if err != nil {
+		return err
+	}
+	wikiArgs := &wiki.WikiCreateParametersV2{Name: converter.String(d.Get("name").(string)), ProjectId: &uuidProject, Type: &wikiType}
+	mappedPath, b := d.GetOk("mappedpath")
+
+	if b {
+		wikiArgs.MappedPath = converter.String(mappedPath.(string))
+	}
+	repositoryId, b := d.GetOk("repository_id")
+	if b {
+		repositoryUuid, err := uuid.Parse(repositoryId.(string))
+		if err != nil {
+			return err
+		}
+		wikiArgs.RepositoryId = &repositoryUuid
+	}
+
+	version, b := d.GetOk("versions")
+	if b {
+		wikiArgs.Version = &git.GitVersionDescriptor{Version: converter.String(version.(string))}
+	}
+
+	CreateWikiArgs := wiki.CreateWikiArgs{WikiCreateParams: wikiArgs}
+	resp, err := clients.WikiClient.CreateWiki(clients.Ctx, CreateWikiArgs)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(resp.Id.String())
+	return resourceWikiRead(d, m)
+}
+
+func resourceWikiRead(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	resp, err := clients.WikiClient.GetWiki(clients.Ctx, wiki.GetWikiArgs{WikiIdentifier: converter.String(d.Id())})
+	if err != nil {
+		return err
+	}
+
+	if resp.Id != nil {
+		d.SetId(resp.Id.String())
+		d.Set("id", *resp.Id)
+	}
+	if resp.Name != nil {
+		d.Set("name", *resp.Name)
+	}
+	if resp.ProjectId != nil {
+		d.Set("project_id", *resp.ProjectId)
+	}
+	if resp.Type != nil {
+		d.Set("type", *resp.Type)
+	}
+	if resp.MappedPath != nil {
+		d.Set("mappedpath", *resp.MappedPath)
+	}
+	if resp.RemoteUrl != nil {
+		d.Set("remoteurl", *resp.RemoteUrl)
+	}
+	if resp.RepositoryId != nil {
+		d.Set("repository_id", *resp.RepositoryId)
+	}
+	if resp.Url != nil {
+		d.Set("url", *resp.Url)
+	}
+	if resp.Versions != nil {
+		d.Set("versions", *resp.Versions)
+	}
+
+	return nil
+}
+
+func resourceWikiUpdate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	requiresUpdate := false
+	var updateParameters wiki.WikiUpdateParameters
+
+	if d.HasChange("name") {
+		updateParameters.Name = converter.String(d.Get("name").(string))
+		requiresUpdate = true
+	}
+	if requiresUpdate {
+		_, err := clients.WikiClient.UpdateWiki(clients.Ctx, wiki.UpdateWikiArgs{
+			WikiIdentifier:   converter.String(d.Id()),
+			UpdateParameters: &updateParameters})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func resourceWikiDelete(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	//  codewiki can be deleted normally, for project wiki the repo needs to be deleted
+	wikiType := wiki.WikiType(d.Get("type").(string))
+	if wikiType == "codeWiki" {
+		_, err := clients.WikiClient.DeleteWiki(clients.Ctx, wiki.DeleteWikiArgs{
+			WikiIdentifier: converter.String(d.Id()),
+			Project:        converter.String(d.Get("project_id").(string))})
+		if err != nil {
+			return err
+		}
+	} else if wikiType == "projectWiki" {
```

Comment:
An error message should pop up instead of removing the parent resource. Each resource is dedicated to its function and anything beyond its control should not be managed